crowbar_framework/app/controllers/installer/upgrades_controller.rb

```diff
@@ -323,9 +323,14 @@ def post_cleanup_repos
           false
         )
 
-        Crowbar::Repository.chef_data_bag_destroy(
-          "#{repository.data_bag_name}/#{repository.id}"
-        )
+        begin
+          Crowbar::Repository.chef_data_bag_destroy(
+            "#{repository.data_bag_name}/#{repository.id}"
+          )
+        rescue Net::HTTPServerException
+          # it is fine to pass if the databag doesn't exist
```

Comment:
Does this really work? According to /usr/lib64/ruby/2.1.0/net/http/responses.rb ```Net::HTTPServerError``` corresponds to errors from the 5xx range. AFAICS ```Crowbar::Repository.chef_data_bag_destroy``` returns 404 Not Found when an item does not exist.

In general I think it's better to catch a more specific error anyway. Just ignoring every Exception that ```Crowbar::Repository.chef_data_bag_destroy``` might throw seems wrong to me.